src/utils/i18n.tsx

```diff
@@ -1,39 +1,34 @@
 // Copyright (c) 2018 Ultimaker B.V.
-import { po } from 'gettext-parser';
 import * as React from 'react';
+import { po } from 'gettext-parser';
 
 import Gettext = require('node-gettext');
 
-// type for available languages
-export type Languages = 'en-US' | 'nl-NL';
-
-// interface for translation items that are passed into initialize
-export interface TranslationListItem {
-    name: Languages
-    source: any
-}
-
 /**
  * Assorted methods for translations.
  */
 export class I18n {
     private static _gt = new Gettext({ debug: false })
 
-    private static _defaultLanguage = 'en-DEV'
+    private static _defaultLanguage = 'en'
 
-    private static _supportedLanguages: Languages[] = ['en-US', 'nl-NL']
+    private static _supportedLanguages: string[] = []
 
     /**
-    * Initialize the Gettext context.
-    */
-    public static async initialize(translations: TranslationListItem[]) {
-        // load the translation files
-        translations.forEach(async translation => (
-            this._loadTranslation(translation.name, translation.source)
-        ));
+     * Initialize the Gettext context.
+     * @param translations The translation keys with files to load.
+     */
+    public static async initialize(translations: { [s: string]: string; }): Promise<void> {
+        // register supported languages
+        Object.keys(translations).forEach((language) => {
+            I18n._supportedLanguages.push(language);
+        });
 
         // set the default locale based on the browser settings
-        I18n._gt.setLocale(I18n.getLocale());
+        const locale = I18n.getLocale();
+        if (I18n.isLanguageSupported(locale)) {
```

Comment:
Shouldn't we have a default language when not supported? Also can't we have multiple locale in the browser? I have for example en-US, nl-NL, pt-BR, en-UK, nl-BE, pt-PT configured in this order

Reply:
The default language is english, this is already handled by the gettext library itself when registering translations. It support multi-locale fine, if there's multiple it just gets the top one from navigator.languages.